templates/types/streaming/nextjs/app/components/ui/chat/hooks/use-config.ts

```diff
@@ -4,27 +4,33 @@ import { useEffect, useMemo, useState } from "react";
 
 export interface ChatConfig {
   chatAPI?: string;
+  embedAPI?: string;
   starterQuestions?: string[];
 }
 
-export function useClientConfig() {
-  const API_ROUTE = "/api/chat/config";
+export function useClientConfig(): ChatConfig {
   const chatAPI = process.env.NEXT_PUBLIC_CHAT_API;
   const [config, setConfig] = useState<ChatConfig>({
     chatAPI,
   });
 
-  const configAPI = useMemo(() => {
-    const backendOrigin = chatAPI ? new URL(chatAPI).origin : "";
-    return `${backendOrigin}${API_ROUTE}`;
+  const backendOrigin = useMemo(() => {
+    return chatAPI ? new URL(chatAPI).origin : "";
   }, [chatAPI]);
 
+  const configAPI = `${backendOrigin}/api/chat/config`;
+  const embedAPI = `${backendOrigin}/api/chat/embed`;
+
   useEffect(() => {
     fetch(configAPI)
       .then((response) => response.json())
       .then((data) => setConfig({ ...data, chatAPI }))
       .catch((error) => console.error("Error fetching config", error));
   }, [chatAPI, configAPI]);
 
-  return config;
+  return {
+    chatAPI,
+    embedAPI,
+    starterQuestions: config.starterQuestions,
```

Comment:
just return `backendOrigin` as `backend` here
```suggestion
    backend
    starterQuestions: config.starterQuestions,
```